dlp/snippets/src/main/java/dlp/snippets/InspectWithStoredInfotype.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package dlp.snippets;
+
+// [START dlp_inspect_with_stored_infotype]
+
+import com.google.cloud.dlp.v2.DlpServiceClient;
+import com.google.privacy.dlp.v2.ContentItem;
+import com.google.privacy.dlp.v2.CustomInfoType;
+import com.google.privacy.dlp.v2.Finding;
+import com.google.privacy.dlp.v2.InfoType;
+import com.google.privacy.dlp.v2.InspectConfig;
+import com.google.privacy.dlp.v2.InspectContentRequest;
+import com.google.privacy.dlp.v2.InspectContentResponse;
+import com.google.privacy.dlp.v2.LocationName;
+import com.google.privacy.dlp.v2.ProjectStoredInfoTypeName;
+import com.google.privacy.dlp.v2.StoredType;
+import java.io.IOException;
+
+public class InspectWithStoredInfotype {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample.
+    // The Google Cloud project id to use as a parent resource.
+    String projectId = "your-project-id";
+    // Specify the stored InfoType to use for inspecting the data.
+    String infoTypeId = "your-info-type-id";
+    // The string to de-identify.
+    String textToDeidentify =
+        "My phone number is (223) 456-7890 and my email address is gary@example.com.";
+    inspectWithStoredInfotype(projectId, infoTypeId, textToDeidentify);
+  }
+
+  //  Inspects the given text using the specified stored infoType detector.
+  public static void inspectWithStoredInfotype(
+      String projectId, String infoTypeId, String textToDeidentify) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the "close" method on the client to safely clean up any remaining background resources.
+    try (DlpServiceClient dlp = DlpServiceClient.create()) {
+
+      // Specify the content to be inspected.
+      ContentItem contentItem = ContentItem.newBuilder().setValue(textToDeidentify).build();
+
+      InfoType infoType = InfoType.newBuilder().setName("GITHUB_LOGINS").build();
```

Comment:
This looks unused. Remove to avoid confusion

Reply:
This is used while creating [CustomInfoType](https://github.com/GoogleCloudPlatform/java-docs-samples/pull/8014/files#diff-0ca3d54b1ba1cba53fab98cef96562be00c1b26d6d274cc699e31dab2e0c0151R66)

Reply:
Hmm, why are we using `GITHUB_LOGINS` here if the infotypeid is passed in as the second argument?

Reply:
```
InfoType infoType = InfoType.newBuilder().setName("GITHUB_LOGINS").build();

// Reference to the existing StoredInfoType to inspect the data.
      StoredType storedType = StoredType.newBuilder()
              .setName(ProjectStoredInfoTypeName.of(projectId, storedInfoTypeId).toString())
              .build();

CustomInfoType customInfoType =
          CustomInfoType.newBuilder().setInfoType(infoType).setStoredType(storedType).build();
```


Here, `GITHUB_LOGINS` is the name of customInfoType and storedInfoTypeId is where we want the new custom infoType to base on.

Reply:
I still don't understand why it's called `GITHUB_LOGINS`. The customer could pass in any infotype right, not just github account emails?

If possible, can we use the same name that's passed in? If not, lets call it something more generic, like `STORED_TYPE`.